cmd/controller-manager/main.go

```diff
@@ -266,4 +289,33 @@ func addMonitoringKeyLabelToOperatorService(ctx context.Context, cfg *rest.Confi
 
 	return nil
 }
- 
\ No newline at end of file
+
+func installMonitoring(ctx context.Context, client client.Client) error {
+
+	params := map[string]string{"Namespace": os.Getenv("NAMESPACE")}
+
+	templateHelper := util.NewTemplateHelper(params)
+
+	for _, template := range templateHelper.TemplateList {
+		resource, err := templateHelper.CreateResource(template)
+		if err != nil {
+			return fmt.Errorf("createResource failed: %s", err)
+		}
+		err = client.Create(ctx, resource)
+		if err != nil {
+			if !kerrors.IsAlreadyExists(err) {
+				return fmt.Errorf("error creating resource: %s", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func hasApplicationMonitoring(config *rest.Config) (bool, error) {
+	dc := discovery.NewDiscoveryClientForConfigOrDie(config)
+	apiVersion := "applicationmonitoring.integreatly.org/v1alpha1"
```

Comment:
I think checking for the ApplicationMonitoring CRD is a bit too narrow. Ideally it would install the service monitor and rules if just the prometheus operator CRDs were installed. 

Same with grafana I guess. Maybe just applying and ignoring the error is more flexible than checking for specific resources?